docs/components/miranum-connect/java-client/miranum-worker.md

```diff
@@ -0,0 +1,113 @@
+│       └── java
+│         ├── adapter
+│         │   ├── in
+```
+
+Having created a class in the package we now want to add such kind of worker. Therefore, we want to check our imports. 
+Make sure to use:
+```java
+import io.miragon.miranum.connect.worker.api.Worker;
+```
+
+In the next step we want to implement a method which acts as a worker. 
+This happens by adding the `@Worker` annotation. You are required to set a type in there. The type matches the corresponding
+type/topic defined in the BPMN activity. 
+
+Your method can be of any return type. Set void if you do not want to store anything back to the scope of the process engine. 
+If you have an object in return it will be stored in a local-context in the process-instance. If you want to use it globally 
+make sure to define the input/output mappings in your process.
+
+```java
+@Worker(type = "my-type") 
+public void doSomething(DoSomethingCommand doSomethingCommand) {
+    doSomethingUseCase.doSomething(doSomethingCommand);
+}
+```
+
+### Commands and UseCases
+As you can see in above examples we deal with Commands and UseCases. These are typically being created in the **ports-in** package
+of our project structure.
+```bash
+│         ├── adapter
+│         │   ├── in
+│         │   └── out
+│         ├── application
+│         │   ├── port
+│         │   │    ├── in
+│         │   │    └── out
+│         │   ├── service
+```
+The method has an input parameter of type `DoSomethingCommand`. This command contains the data which we need from 
+the process in order to execute the worker. (e.g. a list of Strings). 
+
+Whereas the `doSomethingUseCase` is an interface which exists in the same **ports-in** package.
+The actual implementation of that interface can be found in the **service** package. The UseCase implementation could look
+something like this:
+```java
+public class DoSomethingService implements DoSomethingUseCase {
+
+    private final DomainObject domainObject;
+
+    @Override
+    public void doSomething(DoSomethingCommand doSomethingCommand) {
+        domainObject.doSomething(doSomethingCommand.getSomething());
+    }
+}
+```
+
+That is also the point where we interact with our domain object and execute the actual business logic. Keep in ming that 
+in a hexagonal architecture all dependencies point towards the domain-object. On itself it has no dependencies to anything else. 
+
+### Return variables to the process
+As already mentioned in a previous section of this documentation page, variables which are returned by the worker are treated as
+local variables for an activity by default. To change this, [input / output mappings](https://docs.camunda.io/docs/components/concepts/variables/#inputoutput-variable-mappings) 
+are required. (in this case we need the output mapping)
+
+Below you can see an exemplary worker signature with return type `Something`. 
+```java
+@Worker(type = "my-type") 
+public Something doSomething(DoSomethingCommand doSomethingCommand) 
+```
```

Comment:
Ok thats good, didn't see that. Forget about the comment form above. But maybe it really can be better to first show the method with the return type or just show the one with the return type because void is trivial as I mentioned above

Reply:
My thought was to make have the first example of a worker as easy as possible without any kind of return type. 
Hence, I used a separate subchapter to talk about what needs to be taken care of if we want to return something.